src/app/services/module-fetcher.service.spec.ts

```diff
@@ -1,194 +1,126 @@
+    req.flush([]);
+    tick();
   }));
 
-  // Similar mocking done for createNewModule()
-  it('should return an Observable<HttpHeaderResponse> Create', fakeAsync(() => {
+  it('should test createOrUpdateModule id < 1', fakeAsync(() => {
     let url: string = baseURL + '/modules';
-    let response = {
-      resaultCount: 1,
-      response: [
-        {
-          id: 5,
-          subject: 'Java',
-          created: 14454,
-          links: [{ id: 12, contentId: 12, moduleId: 5, affiliation: 'Java' }]
-        }
-      ]
-    };
-    let links: Link[] = [];
-    let input: Module = new Module(null, "CSS", null, links, [], [], [])
-    let output: HttpHeaderResponse;
-
-    service.createOrUpdateModule(input).subscribe(
-      response => {
-        output = response;
-      }
-    );
+    service.createOrUpdateModule(new Module(-1, "CSS", null, [], [], [], [])).subscribe();
     const req = httpTestingController.expectOne(url);
     expect(req.request.method).toEqual("POST");
-    req.flush(response);
+    req.flush(null);
     tick();
+  }));
 
+  it('should test createOrUpdateModule id > 0', fakeAsync(() => {
+    let url: string = baseURL + '/modules/5';
+    service.createOrUpdateModule(new Module(5, "CSS", null, [], [], [], [])).subscribe();
+    const req = httpTestingController.expectOne(url);
+    expect(req.request.method).toEqual("PUT");
+    req.flush(null);
+    tick();
   }));
 
-  //Used to mock delete a module by id
-  it('should return an Observable<Module> delete by id', fakeAsync(() => {
+  it('should test deleteModuleByID', fakeAsync(() => {
     let url: string = baseURL + '/modules/5'
-    let response = {
-      resaultCount: 0,
-      response: []
-    };
-    let output: HttpHeaderResponse;
+    service.deleteModuleByID(5).subscribe();
+    const req = httpTestingController.expectOne(url);
+    expect(req.request.method).toEqual("DELETE");
+    req.flush(null);
+    tick();
+  }));
 
-    service.deleteModuleByID(5).subscribe(
-      response => {
-        output = response;
-      }
-    );
+  it('should test deleteModuleWithSpecificContent', fakeAsync(() => {
+    let url: string = baseURL + '/modules/5?type=unique'
+    service.deleteModuleWithSpecificContent(5).subscribe();
     const req = httpTestingController.expectOne(url);
     expect(req.request.method).toEqual("DELETE");
-    req.flush(response);
+    req.flush(null);
     tick();
+  }));
 
+  it('should test deleteModuleWithContent', fakeAsync(() => {
+    let url: string = baseURL + '/modules/5?type=all'
+    service.deleteModuleWithContent(5).subscribe();
+    const req = httpTestingController.expectOne(url);
+    expect(req.request.method).toEqual("DELETE");
+    req.flush(null);
+    tick();
   }));
 
-  // A getAllFakeData method is needed in the project, so mocking is needed
-  // once again.
-  it('should return an Observable<Module[]> Get all fake data', fakeAsync(() => {
+  it('should test getAllFakeModules', fakeAsync(() => {
```

Comment:
What is fakeAsync used for? Perhaps a comment/documentation explaining such would be helpful.